### PR TITLE
ci: rework ci with github action

### PR DESCRIPTION
**Pull request by Yelinz**
add dependabot, release and test with github actions, removed travis

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+jobs:
+  lint-hbs:
+    name: Lint Handlebars
+    runs-on: [ubuntu-latest]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v1
+
+    - name: Lint Handlebars
+      uses: Borales/actions-yarn@master
```
Comment (anehx) [inline .github/workflows/test.yml]: `borales/actions-yarn` is deprecated. Use `actions/setup-node`

Comment (anehx) [reply .github/workflows/test.yml]: Also, how can you lint withou installing the dependencies?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    - name: Ember tests (floating dependencies)
+      uses: alexlafroscia/actions-ember-testing@v1.0.0
+
+    - name: Install dependencies
+      uses: Borales/actions-yarn@master
+      with: { args: 'install' }
+
+    - name: Ember tests
+      uses: alexlafroscia/actions-ember-testing@v1.0.0
```
Comment (anehx) [inline .github/workflows/test.yml]: Just use `actions/setup-node` with `run: yarn test`

File: package.json [hunk 3]
```diff
@@ -24,8 +24,7 @@
     "lint:js": "eslint .",
     "start": "ember serve",
     "test": "ember test",
-    "test:all": "ember try:each",
-    "precommit": "lint-staged"
```
Comment (anehx) [inline package.json]: Why is this removed?

File: package.json [hunk 4]
```diff
@@ -79,17 +79,31 @@
...
-    "husky": "^0.14.3",
+    "husky": "4.0.10",
     "lint-staged": "^7.0.4",
     "loader.js": "^4.7.0",
     "prettier": "^1.12.0",
     "qunit-dom": "^0.8.0"
   },
   "engines": {
-    "node": "6.* || 8.* || >= 10.*"
+    "node": "8.* || >= 10.*"
```
Comment (anehx) [inline package.json]:
```suggestion
    "node": "10.* || >= 12.*"
```

Node 8.x is not an active LTS anymore

File: package.json [hunk 5]
```diff
@@ -79,17 +79,31 @@
...
   },
   "ember-addon": {
     "configPath": "tests/dummy/config",
     "demoURL": "https://adfinis-sygroup.github.io/ember-validated-form"
+  },
+  "release": {
+      "extends": "@adfinis-sygroup/semantic-release-config"
```
Comment (anehx) [inline package.json]: This is weirdly intended

**Comment by fkm** [comment]
> Please drop all usage of `borales/actions-yarn` and `alexlafroscia/actions-ember-testing` in favor of plain `actions/setup-node` with `run: yarn install` or `run: yarn test`
> 
> See: https://github.com/actions/setup-node

Good to know. I'll update ember-document-merge-service as-well.

File: package.json (hunk 3 above)
Comment (Yelinz) [reply package.json]: It was moved to the bottom of the file

File: .github/workflows/test.yml (hunk 1 above)
Comment (Yelinz) [reply .github/workflows/test.yml]: Changed

File: .github/workflows/test.yml (hunk 2 above)
Comment (Yelinz) [reply .github/workflows/test.yml]: Changed

File: package.json (hunk 4 above)
Comment (Yelinz) [reply package.json]: Changed

File: package.json (hunk 5 above)
Comment (Yelinz) [reply package.json]: Fixed

File: .dependabot/config.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+version: 1
+update_configs:
+  - package_manager: "javascript"
+    directory: "/"
+    update_schedule: "live"
+    version_requirement_updates: "increase_versions"
+    ignored_updates:
+      # Addons that require ember octane (3.14.0+) can't be updated since we
+      # guarantee support for the latest LTS version of ember. Those can be
+      # removed as soon as a new LTS version of ember is released.
```
Comment (anehx) [inline .dependabot/config.yml]: That's not quite true. `ember-power-select` 3.x requires `ember` 3.11.0+. However in this addon we support the last two LTS version which are 3.12 and 3.8 right now.

Comment (Yelinz) [reply .dependabot/config.yml]: Corrected the mistake
